sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/StartSpanOptions.java

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.util.tracing;
+
+import com.azure.core.annotation.Fluent;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents span options that are available before span starts and describe it.
+ */
+@Fluent
+public class StartSpanOptions {
+
+    /**
+     * Type of Span. Only CLIENT and INTERNAL kinds are supported.
+     */
+    public enum Kind {
+        /** Default value. Indicates that the span is used internally. */
+        INTERNAL,
+
+        /**
+         * Indicates that the span covers the client-side wrapper around an RPC or other remote request.
+         */
+        CLIENT,
+    }
+
+    private boolean makeCurrent;
+    private final Kind spanKind;
+    private Map<String, Object> attributes;
+
+    /**
+     * Describes span with given name and INTERNAL kind
+     */
+    public StartSpanOptions() {
+        this(Kind.INTERNAL);
+    }
+
+    /**
+     * Describes span with given name and kind
+     *
+     * @param kind The kind of the span to be created, only INTERNAL and CLIENT are supported.
+     */
+    public StartSpanOptions(Kind kind) {
+        Objects.requireNonNull(kind, "'kind' cannot be null.");
+        this.spanKind = kind;
+        this.attributes = null;
+        this.makeCurrent = false;
+    }
+
+    /**
+     * Sets flag that controls if span should be made current after it's started.
+     * @param makeCurrent flag indicating if span should be made crurent after start..
+     *
+     * @return this instance for chaining.
+     */
+    public StartSpanOptions setMakeCurrent(boolean makeCurrent) {
+        this.makeCurrent = makeCurrent;
+        return this;
+    }
+
+    /**
+     * Sets attribute on span before its started. Such attributes may affect sampling decision.
+     *
+     * @param key attribute key.
+     * @param value attribute value. Note that underlying tracer implementations limit supported value types to
```

Comment:
Good note here 👍 

I don't think we currently log if the user was to pass the value outside of the accepted set. Do you think we should catch if it does and log this?

Reply:
I added a warning log. The consumers of this API are SDKs and they should have tests if attributes are not populated. Rewriting OTel attributes logic here seems like an overkill.